File: php-laravel/lib/app/Models/RequestTimeMapDepartureSearch.php
<?php
/**
 * RequestTimeMapDepartureSearch
 */
namespace app\Models;

/**
 * RequestTimeMapDepartureSearch
 */
class RequestTimeMapDepartureSearch {

    /** @var string $id */
    private $id;

    /** @var \app\Models\Coords $coords */
    private $coords;

    /** @var \app\Models\RequestTransportation $transportation */
    private $transportation;

    /** @var int $travelTime */
    private $travelTime;

    /** @var \DateTime $departureTime */
    private $departureTime;

    /** @var \app\Models\RequestTimeMapProperty[] $properties */
    private $properties;

    /** @var \app\Models\RequestRangeNoMaxResults $range */
    private $range;

}

File: fsharp-functions/OpenAPI/src/model/RequestTimeMapDepartureSearch.fs
namespace OpenAPI.Model

open System
open System.Collections.Generic
open Newtonsoft.Json
open OpenAPI.Model.Coords
open OpenAPI.Model.RequestRangeNoMaxResults
open OpenAPI.Model.RequestTimeMapProperty
open OpenAPI.Model.RequestTransportation

module RequestTimeMapDepartureSearch = 

  //#region RequestTimeMapDepartureSearch

  [<CLIMutable>]
  type RequestTimeMapDepartureSearch = {
    [<JsonProperty(PropertyName = "id")>]
    Id : string;
    [<JsonProperty(PropertyName = "coords")>]
    Coords : Coords;
    [<JsonProperty(PropertyName = "transportation")>]
    Transportation : RequestTransportation;
    [<JsonProperty(PropertyName = "travel_time")>]
    TravelTime : int;
    [<JsonProperty(PropertyName = "departure_time")>]
    DepartureTime : DateTime;
    [<JsonProperty(PropertyName = "properties")>]
    Properties : RequestTimeMapProperty[];
    [<JsonProperty(PropertyName = "range")>]
    Range : RequestRangeNoMaxResults;
  }
  
  //#endregion
  